### ISSUE TITLE
unused_variables - false positive for abstracts

### ISSUE THREAD

**Issue by DGRHicron** [abaplint/abaplint#3092]
```abap
REPORT zfoobar.

CLASS lcl_test DEFINITION ABSTRACT.
  PUBLIC SECTION.
    METHODS test
      RETURNING VALUE(result) TYPE string.
  PROTECTED SECTION.
    METHODS get_foo ABSTRACT
      IMPORTING foo           TYPE string
      RETURNING VALUE(result) TYPE string.

ENDCLASS.

CLASS lcl_test IMPLEMENTATION.
  METHOD test.
    DATA(lv_bar) = `1`.
    result = get_foo( lv_bar ).
  ENDMETHOD.
ENDCLASS.
```

/zfoobar.prog.abap[9, 17]: Variable "foo" not used(unused_variables)

**Comment by larshp** [comment]
yea, well, yea, in this case nothing REDEFINEs `get_foo`, so its not used

guess adding a `allowAbstract` setting might fix the scenario? Just designing something for inheritance is bad IMHO, https://en.wikipedia.org/wiki/Composition_over_inheritance

**Comment by DGRHicron** [comment]
> guess adding a allowAbstract setting might fix the scenario? 
Where is that? I can't find it in schema.

> Just designing something for inheritance is bad IMHO,
Ok, i respect that, but this shouldn't mean that abaplint can throw false positive on it 😉

**Comment by larshp** [comment]
@DGRHicron try again, 2.102.42 published
